src/shell/server.cpp

```diff
@@ -625,6 +629,40 @@ json server::info(std::shared_ptr<module_info const> const & mod_info, pos_info
     return j;
 }
 
+json server::widget_event(std::shared_ptr<module_info const> const & mod_info, pos_info const & pos, json const & message) {
+    json j;
+    try {
+        parse_breaking_at_pos(mod_info->m_id, mod_info, pos);
```

Comment:
I don't think you need `parse_breaking_at_pos` here.  It's main use is auto-completion where we need to know where the current token begins.  But here we already know the position (`pos`?).

Reply:
tbh I found it very difficult to understand the code that translates source positions to info thunks (everything under `report_info` in /src/frontends/lean/interactive.cpp)
, so I tried to make sure that report_widget and widget_event both followed exactly the same log-tree lookup logic as report_info, and since report_info did, update_widget does too.
It seems that when you run parse_breaking_at_pos, you get a different position stored as `m_pos` on the exception args? You need to get the same log tree entry that the report_info step got and I couldn't see how to get that m_pos other than by running parse_breaking_at_pos